tools/block-generator/generator/config.go

```diff
@@ -0,0 +1,323 @@
+// Copyright (C) 2019-2023 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package generator
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+)
+
+// ---- types ----
+
+// TxTypeID is the transaction type.
+type TxTypeID string
+
+const (
+	genesis TxTypeID = "genesis"
+
+	// TX Distribution / ID's
+	paymentTx     TxTypeID = "pay"
+	assetTx       TxTypeID = "asset"
+	applicationTx TxTypeID = "appl"
+	//keyRegistrationTx TxTypeID = "keyreg"
+
+	// Payment TX Distribution / ID's
+	paymentAcctCreateTx TxTypeID = "pay_create"
+	paymentPayTx        TxTypeID = "pay_pay"
+
+	// Asset TX Distribution / ID's
+	assetCreate  TxTypeID = "asset_create"
+	assetDestroy TxTypeID = "asset_destroy"
+	assetOptin   TxTypeID = "asset_optin"
+	assetXfer    TxTypeID = "asset_xfer"
+	assetClose   TxTypeID = "asset_close"
+
+	// App kind TX Distribution / ID's don't exist because these are flattened
+	// into weights across (app kinds) X (app tx type)
+
+	// App Swap TX Distribution / ID's
+	appSwapCreate TxTypeID = "app_swap_create"
+	appSwapUpdate TxTypeID = "app_swap_update"
+	appSwapDelete TxTypeID = "app_swap_delete"
+	appSwapOptin  TxTypeID = "app_swap_optin"
+	appSwapCall   TxTypeID = "app_swap_call"
+	appSwapClose  TxTypeID = "app_swap_close"
+	appSwapClear  TxTypeID = "app_swap_clear"
+
+	// App Boxes TX Distribution / ID's
+	appBoxesCreate TxTypeID = "app_boxes_create"
+	appBoxesUpdate TxTypeID = "app_boxes_update"
+	appBoxesDelete TxTypeID = "app_boxes_delete"
+	appBoxesOptin  TxTypeID = "app_boxes_optin"
+	appBoxesCall   TxTypeID = "app_boxes_call"
+	appBoxesClose  TxTypeID = "app_boxes_close"
+	appBoxesClear  TxTypeID = "app_boxes_clear"
```

Comment:
In the next PR will you remove the configs that aren't used for anything yet?

Reply:
Not in the next PR, but in the final PR of the series once it's clear what's actually done. I'm suspecting I'll get rid of:
* `*update`
* `*delete`
* `*close`
* `*clear`

Reply:
but I can comment out now and uncomment as needed, LMK

Reply:
no need, just curious